### PR TITLE
17.4: Convert symbol-search crash to NFW

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Backport of https://github.com/dotnet/roslyn/pull/65279 to 17.4

### CODE REVIEW

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs
Comment (dibarbet) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:565]: same q on the `_service` here

**Review by dibarbet — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:565]: Answered in other PR.  this allows the tests to actually validate tehse codepaths without actually invoking FatalError (which kills tests).

**Review by ryzngard — APPROVED** [review]

**Comment by arkalyanms** [comment]
This is now targeting 17.4.2 servicing. So let's keep this on deck until the 17.4.1 clears out and merge after.

**Comment by CyrusNajmabadi** [comment]
@arkalyanms Sounds good.  LMK when this can go in.

**Comment by CyrusNajmabadi** [comment]
@arkalyanms are we still waiting on this?

**Comment by arkalyanms** [comment]
Yes the 17.4.2 branch has not opened up yet (on Nov 21st). But this has soft approval, so should be fine to merge here and insert into VS when time comes.

cc: @jasonmalinowski

**Pull request merged**
